src/test/rgw/amqp_mock.cc

```diff
@@ -61,6 +61,8 @@ bool REPLY_ACK(true);
 
 using namespace amqp_mock;
 
+extern "C" {
```

Comment:
no need to guard the definitions of these `struct`s. how about putting `extern "C"` at line 107?